RegistrySubscriptionManager/src/test/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionServiceTest.java

```diff
@@ -1,145 +1,295 @@
+import eu.neclab.ngsildbroker.commons.subscriptionbase.IntervalNotificationHandler;
+import eu.neclab.ngsildbroker.commons.tools.HttpUtils;
+import eu.neclab.ngsildbroker.registry.subscriptionmanager.repository.RegistrySubscriptionInfoDAO;
+import eu.neclab.ngsildbroker.registry.subscriptionmanager.service.RegistrySubscriptionService;
+import groovy.util.logging.Slf4j;
+
+import com.github.jsonldjava.core.JsonLdOptions;
+import com.github.jsonldjava.utils.JsonUtils;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@QuarkusTest
+@Slf4j
+@TestMethodOrder(OrderAnnotation.class)
+@TestProfile(CustomProfile.class)
+public class SubscriptionServiceTest {
+
+	@Mock
+	SubscriptionRequest subscriptionRequest;
+
+	@Mock
+	IntervalNotificationHandler intervalHandlerREST;
+
+	@Mock
+	HashMap<String, Subscription> subscriptionId2Subscription;
+
+	@Inject
+	RegistrySubscriptionService subscriptionService;
+
+	@Mock
+	NotificationParam param;
+
+	@Mock
+	Subscription subscriptionMock;
+
+	@Mock
+	IntervalNotificationHandler intervalNotificationHandler;
+
+	@Inject
+	RegistrySubscriptionInfoDAO subService;
+
+	ArrayListMultimap<String, String> multimaparr = ArrayListMultimap.create();
+
+	// resolved payload of a subscription.
+
+	String payload = "{\r\n  \"https://uri.etsi.org/ngsi-ld/entities\" : [ "
+			+ "{\r\n    \"@type\" : [ \"https://uri.etsi.org/ngsi-ld/default-context/Vehicle\" ]"
+			+ "\r\n  } ],\r\n  \"@id\" : \"urn:ngsi-ld:Subscription:173223\","
+			+ "\r\n  \"https://uri.etsi.org/ngsi-ld/notification\" : [ "
+			+ "{\r\n    \"https://uri.etsi.org/ngsi-ld/endpoint\" : [ "
+			+ "{\r\n      \"https://uri.etsi.org/ngsi-ld/accept\" : [ "
+			+ "{\r\n        \"@value\" : \"application/json\""
+			+ "\r\n      } ],\r\n      \"https://uri.etsi.org/ngsi-ld/uri\" : [ {"
+			+ "\r\n        \"@value\" : \"http://localhost:8080/acc\"\r\n      } ]"
+			+ "\r\n    } ]\r\n  } ],\r\n  \"@type\" : [ \"https://uri.etsi.org/ngsi-ld/Subscription\" ]" + "\r\n}";
+
+	String payload1 = "{\r\n  \"https://uri.etsi.org/ngsi-ld/entities\" : [ "
+			+ "{\r\n    \"@type\" : [ \"https://uri.etsi.org/ngsi-ld/default-context/Vehicle\" ]"
+			+ "\r\n  } ],\r\n  \"@id\" : \"urn:ngsi-ld:Subscription:173224\","
+			+ "\r\n  \"https://uri.etsi.org/ngsi-ld/notification\" : [ "
+			+ "{\r\n    \"https://uri.etsi.org/ngsi-ld/endpoint\" : [ "
+			+ "{\r\n      \"https://uri.etsi.org/ngsi-ld/accept\" : [ "
+			+ "{\r\n        \"@value\" : \"application/json\""
+			+ "\r\n      } ],\r\n      \"https://uri.etsi.org/ngsi-ld/uri\" : [ {"
+			+ "\r\n        \"@value\" : \"http://localhost:8080/acc\"\r\n      } ]"
+			+ "\r\n    } ]\r\n  } ],\r\n  \"@type\" : [ \"https://uri.etsi.org/ngsi-ld/Subscription\" ]" + "\r\n}";
+
+	/**
+	 * this method is used to test create subscription test
+	 */
+	@Test
+	@Order(1)
+	public void serviceTest() throws ResponseException {
```

Comment:
Please correct test case name as per the operation.

Reply:
Done